### DISCUSSION TITLE
Ender 3 S1 bed size and tramming

### DISCUSSION THREAD

**Discussion by phoenixflare** [mriscoc/Ender3V2S1#634] [Q&A] [answered]
I just installed the newest version of the firmware on the S1 and noticed that the bed size is incorrect. Changing it to the proper 235x235 and setting the offsets worked.
That led up to the issue where the bed leveling was only doing the front left part with a portion of the right and back part of the bed being ignored. Changing the mesh max xy fixed that but then I decided to try out the tramming

The tramming, even on default, seems to be about 5-10mm off the screw to the left and towards the front. Meaning even the default settings had them over the wrong points. I've been trying to change that in the configuration to build a new profile but while I managed to get the Back and Right points correct, the Front and Left points are always stuck and no amount of changing to the TRAMMING_POINT_XY seems to have fixed it. 
#define LCD_BED_TRAMMING gives the "DWIN_LCD_PROUI does not support LCD_BED_TRAMMING." error
#define ASSISTED_TRAMMING gives the "region `FLASH' overflowed by 1076 bytes" error.

**Comment by TheWebMachine** [comment]
I ran into something similar! After changing the Physical Bed parameters, you need to reboot the printer for those values to take effect for tramming and other wizards. Mesh seems to be the only thing that updates physical sizes and limits in real time before the reboot.

**Reply by phoenixflare** [reply]
Definitely needed a power cycle for the tramming locations to update. Kind of weird that it needs it even though I was setting the properties in the firmware.
It updated the points and now both the left and right points are too close to the edge instead of just the left. But that might be cause I increased the max_x value as well so it is now calculating the right point based on that

**Reply by mriscoc** [reply]
I just tested, changing the mesh insets has an immediate effect on the tramming wizard. But if you change bed size, you need to update the mesh insets before of launch the tramming wizard because it uses the mesh corners as tramming points.

**Reply by TheWebMachine** [reply]
Yes, changing insets is immediately reflected in tramming, but *not* changes to physical bed size and min-max values. Those still require a reboot to take effect in tramming. Even when I changed physical and then mesh insets, still had to reboot to get tramming on new physical size. Changing only insets or other mesh vars is immediately reflected in tramming.

**Reply by mriscoc** [reply]
Changing the min/max values and bed size without changing mesh insets has that effect, you need always to update the mesh insets to have the tramming points updated.

**Reply by phoenixflare** [reply]
Can confirm the issue was the Mesh insets. For reference, the S1 I used 27.5, 35 and 202.5, 202.5.
A bit of a pain that the bed can't really go forward enough for the probe. It barely reaches after setting the min and max to 250, 245.

**Comment by mriscoc** [comment, marked as answer]
https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides

**Reply by phoenixflare** [reply]
I already saw that. Also checked the https://github.com/mriscoc/Ender3V2S1/wiki/Bed-tramming-menu to see if there was anything before posting this.
